src/UNL/MediaHub/DefaultHomepage.php

```diff
@@ -22,12 +22,54 @@ function getCacheKey()
 
     function run()
     {
+        
+    }
+    
+    function getTopMedia()
+    {
         $options = array(
-            'limit'   => 6,
+            'limit'   => null,
             'orderby' => 'popular_play_count'
         );
-        $this->top_media = new UNL_MediaHub_MediaList($options);
-        $this->top_media->run();
+        $top_media = new UNL_MediaHub_MediaList($options);
```

Comment:
The expensive run time filter is the 2 nested for loops. As a public method, people may use this method more than once in the future. Any time we can avoid having to do expensive computation/loops, we should.